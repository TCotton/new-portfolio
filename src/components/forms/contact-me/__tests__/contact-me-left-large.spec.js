import React from 'react';
import renderer from 'react-test-renderer';
import ContactMeLeftLarge from '../contact-me-left-large';

describe('ContactMeLeftLarge', () => {

	let component, instance;

	beforeEach( () => {
		component = renderer.create(
			<ContactMeLeftLarge />
		);
		instance = component.getInstance();
	});

	it('renders correctly', () => {
		const tree = renderer
			.create(<ContactMeLeftLarge />)
			.toJSON()
		expect(tree).toMatchSnapshot();
	})

	it('should call shouldComponentUpdate', () => {
		const shouldUpdate = instance.shouldComponentUpdate();
		expect(shouldUpdate).toBe(true);
	});
})
