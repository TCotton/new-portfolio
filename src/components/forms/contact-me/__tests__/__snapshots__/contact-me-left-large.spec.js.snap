// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMeLeftLarge renders correctly 1`] = `
<DocumentFragment>
  <div
    class="contact-left-large"
  >
    <div
      class="hidden-desktop"
    >
      <h4
        class="content-title"
      >
        Get in touch left
      </h4>
      <p>
        Andy Walpole
        <br />
        Web Developer
      </p>
      <p>
        <a
          class="underline"
          href="mailto:me@andywalpole.me?Subject=Contact%20from%20portfolio"
        >
          me@andywalpole.me
        </a>
      </p>
    </div>
    <h3
      class="content-title"
    >
      Send a message
    </h3>
    <p
      class="contact-form-success none"
    >
      Thank you for taking the time to fill out the form. I will contact you as soon as I can!
    </p>
    <p
      class="contact-form-failure block"
    >
      The form has not been submitted because of errors. Please review the form error messages and click submit again
    </p>
    <form
      name="contactForm"
      novalidate=""
    >
      <span
        class="form-division-blocks"
      >
        <label
          class="visuallyhidden"
          for="name"
        >
          Your full name
        </label>
        <input
          class="form-input"
          data-testid="name"
          id="name"
          maxlength="64"
          name="name"
          placeholder="Your full name"
          required=""
          type="text"
          value=""
        />
      </span>
      <span
        class="form-division-blocks"
      >
        <label
          class="visuallyhidden"
          for="email"
        >
          Your email address
        </label>
        <input
          autocapitalize="off"
          autocorrect="off"
          class="form-input"
          data-testid="email"
          id="email"
          maxlength="64"
          name="email"
          placeholder="Your email address"
          required=""
          type="email"
          value=""
        />
      </span>
      <span
        class="form-division-blocks"
      >
        <label
          class="visuallyhidden"
          for="message"
        >
          Please write a message
        </label>
        <textarea
          class="form-textarea"
          cols="10"
          data-testid="message"
          id="message"
          maxlength="1000"
          name="message"
          placeholder="Please write a message"
          required=""
          rows="10"
        />
      </span>
      <span>
        <label
          class="hide"
          for="zipcode"
        >
          zipcode
        </label>
        <input
          data-testid="zipcode"
          id="zipcode"
          name="zipcode"
          placeholder="zipcode"
          type="text"
          value=""
        />
      </span>
      <input
        class="button-contact"
        data-testid="addBtn"
        type="submit"
        value="Send message"
      />
    </form>
  </div>
</DocumentFragment>
`;
